lighthouse-core/lib/page-functions.js

```diff
@@ -225,6 +225,59 @@ function getNodeSelector(node) {
   return parts.join(' > ');
 }
 
+/**
+ * This function checks if an element or an ancestor of an element is `position:fixed`.
+ * In addition we ensure that the element is capable of behaving as a `position:fixed`
+ * element, checking that it lives within a scrollable ancestor.
+ * @param {HTMLElement} element
+ * @return {boolean}
+ */
+/* istanbul ignore next */
+function isPositionFixed(element) {
+  /**
+   * @param {HTMLElement} element
+   * @return {boolean}
+   */
+  function hasScrollableAncestor(element) {
+    let currentEl = element.parentElement;
+    while (currentEl) {
+      if (currentEl.scrollHeight > currentEl.clientHeight) {
+        const yScroll = getStyleAttrValue(currentEl, 'overflowY');
+        // In case where we reach outermost element (`<html>`), default computed value of visible
+        // will be scrollable.
+        if ((yScroll === 'scroll' || yScroll === 'auto') ||
+        (yScroll === 'visible' && currentEl.parentElement === null)
```

Comment:
now that I'm thinking about this more, I can't really think of a situation other than the document isn't at least the window height where this function will ever return false... is that expected it's fairly rare?

maybe we could add some comments to the jsdoc of `hasScrollableAncestor` describing the concrete examples we need this to watchout for?

Reply:
You're completely right, that really is the only case where this will fail so doing lots of unneeded work. Will modify this to only look for that condition.